unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/ChannelLinkDetector.cs

```diff
@@ -0,0 +1,98 @@
+using DCL.Chat.Channels;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace DCL.Chat.HUD
+{
+    public class ChannelLinkDetector : MonoBehaviour, IPointerClickHandler
+    {
+        [SerializeField] internal TMP_Text textComponent;
+
+        internal string currentText;
+        internal bool hasNoParseLabel;
+        internal List<string> channelsFoundInText = new ();
+        private bool isAllowedToCreateChannels = true;
+
+        private void Awake()
+        {
+            if (textComponent == null)
+                return;
+
+            textComponent.OnPreRenderText += OnTextComponentPreRenderText;
+
+            if (Environment.i != null
+                && Environment.i.serviceLocator.Get<IChannelsFeatureFlagService>() != null)
+            {
+                var channelsFeatureFlagService = Environment.i.serviceLocator.Get<IChannelsFeatureFlagService>();
+                isAllowedToCreateChannels = channelsFeatureFlagService.IsChannelsFeatureEnabled();
+            }
+        }
+
+        private void OnDestroy()
+        {
+            if (textComponent == null)
+                return;
+
+            textComponent.OnPreRenderText -= OnTextComponentPreRenderText;
+        }
+
+        public void OnPointerClick(PointerEventData eventData)
+        {
+            if (!isAllowedToCreateChannels) return;
+
+            if (eventData.button != PointerEventData.InputButton.Left) return;
+            string clickedLink = GetChannelLinkByPointerPosition(eventData.position);
+
+            if (!ChannelUtils.IsAChannel(clickedLink)) return;
+            DataStore.i.channels.channelJoinedSource.Set(ChannelJoinedSource.Link);
+            DataStore.i.channels.currentJoinChannelModal.Set(clickedLink.ToLower(), true);
+        }
+
+        private void OnTextComponentPreRenderText(TMP_TextInfo textInfo)
+        {
+            if (currentText == textComponent.text) return;
+            if (!isAllowedToCreateChannels) return;
+
+            hasNoParseLabel = textInfo.textComponent.text.ToLower().Contains("<noparse>");
+            CoroutineStarter.Start(RefreshChannelPatterns());
```

Comment:
We deprecated coroutines a while ago, switch to a unitask!

Reply:
done